app/src/main/java/com/example/project/MapsFragment.kt

```diff
@@ -218,12 +235,75 @@ class MapsFragment : Fragment(), OnMapReadyCallback {
             }
 
         })
+
+        // search for place when user presses enter
+        searchBar.setOnSearchActionListener(object : MaterialSearchBar.OnSearchActionListener {
+            override fun onSearchStateChanged(enabled: Boolean) {
+                val searchState = if (enabled) "enabled" else "disabled"
+                Log.i(Constants.TAG, "Search $searchState")
+            }
+
+            override fun onSearchConfirmed(text: CharSequence?) {
+                if (text?.isNotEmpty()!!){ startSearch(text.toString())}
+            }
+
+            override fun onButtonClicked(buttonCode: Int) {
+            }
+
+        })
+
+        // remove marker when x is clicked
+        searchBar.findViewById<View>(R.id.mt_clear).setOnClickListener {
+            searchBar.closeSearch()
+            searchBar.clearSuggestions()
+            Log.i(Constants.TAG, "${searchedPlaceMarker?.title}")
+            searchedPlaceMarker?.remove()
+            lastKnownLocation?.latitude?.let { it1 -> lastKnownLocation?.longitude?.let { it2 ->
+                LatLng(it1,
+                    it2
+                )
+            } }
+
+        }
+    }
+
+    private fun startSearch(text: String){
+        viewModel.findPlaceFromTextSearch(text,Constants.INPUT_TYPE,Constants.FIELDS,getString(R.string.google_maps_key))
+        viewModel.findPlaceFromTextResponse.observe(this,{
+            if (it != null) {
+                val response = it.candidates[0]
+                val latLng = LatLng(response.geometry.location.lat, response.geometry.location.lng)
+                map?.moveCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        latLng, Constants.DEFAULT_ZOOM.toFloat()
+                    )
+                )
+                searchedPlaceMarker = map?.addMarker(
+                    MarkerOptions().position(latLng)
+                        .title(response.name)
+                )
+
+                map?.setOnMarkerClickListener { marker ->
+                    if (marker.title == searchedPlaceMarker?.title){
+                        response.opening_hours?.open_now?.let { it1 ->
```

Comment:
There are 2 issues here in my opinion
- `it1` is not a meaningful name. Try a more semantic one that helps the code reader understand what is the purpose of that variable / parameter. Trust me, this is really important; not less important than the coding itself.
- With this approach, restaurant whose value of `opening_hours` is `null` will not trigger the details card (panel) as `setUpCard` never get calls. That is not what we desire here, I think. Instead, from my perspective, we should handle this case by doing something like the following: if there is `opening_hours`, display it correspondingly; if there's no `opening_hours`, still set up the card, but let the text of "Open/Close" blank.